tmt/steps/execute/internal.py

```diff
@@ -8,64 +8,29 @@
 
 import tmt
 import tmt.utils
-from tmt.steps.execute import TEST_OUTPUT_FILENAME
+from tmt.steps.execute import TEST_OUTPUT_FILENAME, Script
 from tmt.steps.provision import DEFAULT_RSYNC_OPTIONS
 from tmt.steps.provision.local import GuestLocal
 
-REBOOT_VARIABLES = (
-    "TMT_REBOOT_COUNT",
-    "REBOOTCOUNT",
-    "RSTRNT_REBOOTCOUNT")
-REBOOT_SCRIPT_PATHS = (
-    "/usr/local/bin/rstrnt-reboot",
-    "/usr/local/bin/rhts-reboot",
-    "/usr/local/bin/tmt-reboot")
+# Script handling reboots, in restraint compatible fashion
+tmt_reboot = Script("/usr/local/bin/tmt-reboot",
+                    aliases=[
+                        "/usr/local/bin/rstrnt-reboot",
+                        "/usr/local/bin/rhts-reboot"],
+                    variables=[
+                        "TMT_REBOOT_COUNT",
+                        "REBOOTCOUNT",
+                        "RSTRNT_REBOOTCOUNT"]
+                    )
+
+# Script for archiving a file, usable for BEAKERLIB_COMMAND_SUBMIT_LOG
+tmt_file_submit = Script("/usr/local/bin/tmt-file-submit")
```

Comment:
I'd maybe use upper-case for these two names since they are module-level "constants", and also I'd add a suffix like `_SCRIPT`, to denote they label scripts.

That's all, the patch looks very good to me, it looks like things got much more transparent.